eradiate/scenes/integrators.py

```diff
@@ -0,0 +1,181 @@
+"""
+
+
+@attr.s
+class MonteCarloIntegrator(Integrator):
+    """Base class for integrator elements wrapping kernel classes
+    deriving from
+    :class:`eradiate.kernel.render.MonteCarloIntegrator <mitsuba.render.MonteCarloIntegrator>`.
+
+    .. warning::
+       This class should not be instantiated.
+    """
+    max_depth = attr.ib(
+        default=None,
+        converter=attr.converters.optional(int)
+    )
+
+    rr_depth = attr.ib(
+        default=None,
+        converter=attr.converters.optional(int)
+    )
+
+    hide_emitters = attr.ib(
+        default=None,
+        converter=attr.converters.optional(bool)
+    )
+
+    def kernel_dict(self, ref=True):
+        result = {self.id: {}}
+
+        if self.max_depth is not None:
+            result[self.id]["max_depth"] = self.max_depth
+        if self.rr_depth is not None:
+            result[self.id]["rr_depth"] = self.rr_depth
+        if self.hide_emitters is not None:
+            result[self.id]["hide_emitters"] = self.hide_emitters
+
+        return result
+
+
+# Add doctrings
+MonteCarloIntegrator.__doc__ += f"""
+.. rubric:: Constructor arguments / instance attributes
+
+{_monte_carlo_integrator_params_docs}
+"""
+
+
+@IntegratorFactory.register("path")
+@attr.s
+class PathIntegrator(MonteCarloIntegrator):
+    """A thin interface to the `path tracer kernel plugin <https://eradiate-kernel.readthedocs.io/en/latest/generated/plugins.html#path-tracer-path>`_.
+
+    This integrator samples paths using random walks starting from the sensor.
+    It supports multiple scattering and does not account for volume
+    interactions.
+    """
+
+    def kernel_dict(self, ref=True):
+        result = super(PathIntegrator, self).kernel_dict(ref)
+        result[self.id]["type"] = "path"
+        return result
+
+
+PathIntegrator.__doc__ += f"""
+.. rubric:: Constructor arguments / instance attributes
+
+{_monte_carlo_integrator_params_docs}
+"""
+
+
+@IntegratorFactory.register("volpath")
+@attr.s
+class VolPathIntegrator(MonteCarloIntegrator):
+    """A thin interface to the volumetric path tracer kernel plugin.
+
+    This integrator samples paths using random walks starting from the sensor.
+    It supports multiple scattering and accounts for volume interactions.
+    """
+
+    def kernel_dict(self, ref=True):
+        result = super(VolPathIntegrator, self).kernel_dict(ref)
+        result[self.id]["type"] = "volpath"
+        return result
+
+
+VolPathIntegrator.__doc__ += f"""
+.. rubric:: Constructor arguments / instance attributes
+
+{_monte_carlo_integrator_params_docs}
+"""
+
+
+@IntegratorFactory.register("volpathmis")
+@attr.s
+class VolPathMISIntegrator(MonteCarloIntegrator):
+    """A thin interface to the volumetric path tracer (with spectral multiple
+    importance sampling) kernel plugin
+    :cite:`Miller2019NullscatteringPathIntegral`.
+    """
+
+    use_spectral_mis = attr.ib(
+        default=None,
```

Comment:
Why is the default not `False` ?

Reply:
`None` adds no entry to the kernel dictionary and lets Mitsuba apply its own defaults. If I apply a default and Mitsuba changes its own, mine will remain unchanged.

Reply:
Ahhhh